src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -396,26 +390,19 @@ oop MemAllocator::finish(HeapWord* mem) const {
 }
 
 oop ObjAllocator::initialize(HeapWord* mem) const {
-  mem_clear(mem);
+  mem_clear(mem, instanceOopDesc::base_offset_in_bytes());
   return finish(mem);
 }
 
-MemRegion ObjArrayAllocator::obj_memory_range(oop obj) const {
-  if (_do_zero) {
-    return MemAllocator::obj_memory_range(obj);
-  }
-  ArrayKlass* array_klass = ArrayKlass::cast(_klass);
-  const size_t hs = arrayOopDesc::header_size(array_klass->element_type());
-  return MemRegion(cast_from_oop<HeapWord*>(obj) + hs, _word_size - hs);
-}
-
 oop ObjArrayAllocator::initialize(HeapWord* mem) const {
   // Set array length before setting the _klass field because a
   // non-null klass field indicates that the object is parsable by
   // concurrent GC.
   assert(_length >= 0, "length should be non-negative");
   if (_do_zero) {
-    mem_clear(mem);
+    ArrayKlass* array_klass = ArrayKlass::cast(_klass);
```

Comment:
Do you need this `array_klass`? If this is for asserts, then it should be done as assert.